src/utils/form.ts

```diff
@@ -12,3 +14,148 @@ export const hiddenInputStyle = `
   -webkit-appearance: none !important;
   z-index: -1 !important;
 `;
+
+interface FormDataEvent extends Event {
+  formData: FormData;
+}
+
+/**
+ * Defines interface for components that are form-compatible.
+ *
+ * Along with the interface, use the matching form utils to help set up the component behavior.
+ */
+export interface FormAssociated<T = any> {
```

Comment:
Should we rename this interface something more calcitey? Like `CalciteFormComponent`?

Reply:
Or maybe just `FormComponent` and `LabelComponent`?

Reply:
Since this interface defines attributes common to all kinds of _inputs_ perhaps it should be `InputComponent` or `FormInputComponent`?